src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
+    private void checkIndex(int index) {
+        if (size != 0 && (index >= size || index < 0)) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index
+                    + " out of bounds for length " + size);
+        }
+    }
+
+    public void capacityIncrease() {
+        Object[] newArray = new Object[size + size / 2];
+        System.arraycopy(elements,0, newArray, 0,elements.length);
+        elements = newArray;
+    }
+
     @Override
     public void add(T value) {
+        if (size >= DEFAULT_CAPACITY) {
+            capacityIncrease();
+        }
```

Comment:
so after we add 10 elements, we will increase capacity for each addition
can we change this behaviour?